### DISCUSSION TITLE
Information about database schema

### DISCUSSION THREAD

**Discussion by mskyttner** [duckdb/duckdb#2790] [Q&A] [answered]
If I enter "show lineitems;" in the demo at https://duckdb.org/demo, or if I use

```bash
duckdb tpch_01.duckdb -json 'show lineitem;'
```
I can see information about a specific table.

Why does the "show" stament work? Does it call a PRAGMA?

There are PRAGMAs available for returning information about tables in the current database:

```
PRAGMA database_list;
PRAGMA show_tables;
PRAGMA table_info('table_name');
```
My question is how I can query the results returned from a PRAGMA statement such as table_info, something like `select name from (PRAGMA table_info('lineitems'));`? 

Or could the results from the PRAGMA be loaded into a temp table or view so it can then queried with SQL?

**Comment by Mytherin** [comment, marked as answer]
You can use the `pragma_table_info` function:

`SELECT * FROM pragma_table_info('lineitem');`

**Reply by mskyttner** [reply]
I just tried this statement which enumerates table and column names in the current database, which is what I was looking for:

`SELECT table_name, column_name FROM information_schema_columns() WHERE table_name != 'sqlite_master';`

**Reply by Mytherin** [reply]
The `information_schema_columns` function has been deprecated and removed a few versions ago. You will need to use `information_schema.columns` in the latest version, e.g.:

```sql
SELECT table_name, column_name FROM information_schema.columns; 
```

**Reply by mskyttner** [reply]
Ok, thanks!
